extensions/amp-access/0.1/amp-access.js

```diff
@@ -594,9 +601,10 @@ export class AccessService {
     }
 
     log.fine(TAG, 'Start login');
-    const urlPromise = this.buildUrl_(assert(this.config_.login,
-        'Login URL is not configured'), /* useAuthData */ true);
-    this.loginPromise_ = this.openLoginDialog_(urlPromise).then(result => {
+    assert(this.config_.login, 'Login URL is not configured');
+    // Login URL should always be available at this time.
+    const loginUrl = assert(this.loginUrl_);
```

Comment:
@cramforce In theory nothing should block loginUrl resolution. I even initially resolve loginUrl without authorization response and then again after authorization response has been received. Otherwise, I don't have any other way to determine whether or not open(blank) + location set would succeed.